mlflow/store/model_registry/abstract_store.py

```diff
@@ -830,3 +841,91 @@ def search_prompt_versions(
             "This method is only available in Unity Catalog registries.",
             INVALID_PARAMETER_VALUE,
         )
+
+    def link_prompts_to_trace(self, prompt_versions: list[PromptVersion], trace_id: str) -> None:
+        """
+        Link multiple prompt versions to a trace.
+
+        Default implementation sets a tag on the trace. Stores can override with custom behavior.
+
+        Args:
+            prompt_versions: List of PromptVersion objects to link.
+            trace_id: Trace ID to link to each prompt version.
+        """
+        from mlflow.tracking import _get_store as _get_tracking_store
+
+        tracking_store = _get_tracking_store()
+
+        with self._prompt_link_lock:
+            try:
+                trace_info = tracking_store.get_trace_info(trace_id)
+                if not trace_info:
+                    raise MlflowException(
+                        f"Could not find trace with ID '{trace_id}' to which to link prompts.",
+                        error_code=ErrorCode.Name(RESOURCE_DOES_NOT_EXIST),
+                    )
```

Comment:
try-catch suppresses this error.

Reply:
Great catch - we shouldn't be try/catching this method. Fixed!